src/components/Timer/Timer.js

```diff
@@ -31,73 +31,59 @@ const formats = {
   s: 's',
 }
 
-const getFormat = memoize(format =>
-  ['y', 'M', 'd', 'h', 'm', 's'].map(symbol => formats[format].includes(symbol))
-)
-
-// Remove “0” units on the left side
-function removeLeftZeros(units) {
-  // keep at least one unit, even if its value is 0
-  const lastZeroIndex = units.map(u => u[1]).lastIndexOf(0)
-  for (let i = 0; i < units.length && i < lastZeroIndex; i++) {
-    if (units[i][1] !== 0 && units[i][1] !== null) {
-      return units
-    }
-    units[i][1] = null
-  }
-  return units
+const unitNames = {
+  y: 'years',
+  M: 'months',
+  d: 'days',
+  h: 'hours',
+  m: 'minutes',
+  s: 'seconds',
 }
 
-function getTime(start, end, format, showEmpty) {
-  const [
-    showYears,
-    showMonths,
-    showDays,
-    showHours,
-    showMinutes,
-    showSeconds,
-  ] = getFormat(format)
+const getFormat = memoize(format =>
+  ['y', 'M', 'd', 'h', 'm', 's'].reduce(
+    (units, symbol) =>
+      formats[format].includes(symbol) ? [...units, unitNames[symbol]] : units,
+    []
+  )
+)
 
-  const period = end ? [end, new Date()] : [new Date(), start]
+function getTime(start, end, format, showEmpty, maxUnits) {
+  const date1 = end || new Date()
+  const date2 = end ? new Date() : start
```

Comment:
Not sure, but I think this shoudl use `start`?

```suggestion
  const date2 = start ? new Date() : start
```

Reply:
They are exclusive: if `end` is set, `start` is always `new Date()`.